### PR TITLE
fix(storybook): change output config

### PR DESCRIPTION
**Pull request by targeral**
# PR Details

Modify the webpack configuration of the running storybook to override the output configuration of the storybook with the webpackConfig.output configuration inside modern.js.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts [hunk 1]
```diff
@@ -144,6 +145,10 @@ export const getCustomWebpackConfigHandle: any = ({
 }) => {
   const { RULE, PLUGIN, ONE_OF } = CHAIN_ID;
   const { appDirectory } = appContext;
+
+  // Manual configuration `output.path = 'storybook-static'`;
+  modernConfig.output.path = './dist/storybook-static';
```
Comment (chenjiahan) [inline packages/cli/plugin-storybook/src/features/utils/webpackConfig.ts:150]: 感觉在这里修改 modernConfig 不是很好维护，能定义在 config hook 里吗

File: packages/cli/plugin-storybook/src/index.ts [hunk 2]
```diff
@@ -27,6 +27,10 @@ export default (): CliPlugin => ({
             '@modern-js/runtime/plugins': pluginsExportsUtils.getPath(),
           },
         },
+        output: {
```
Comment (chenjiahan) [inline packages/cli/plugin-storybook/src/index.ts:30]: 是不是要加个判断条件嘞，只在构建 storybook 时设置

Comment (targeral) [reply packages/cli/plugin-storybook/src/index.ts:30]: 的确，不过除了通过检测 process.argv 上的参数以外，貌似没有其他方式判断。额外增加环境变量倒是能解决，不过加环境变量的话，感觉还是要慎重点吧。所以感觉之前的方式可能会好点，起码在我们后面能够提供一些信息来区别运行环境之前是这样。

Comment (targeral) [reply packages/cli/plugin-storybook/src/index.ts:30]: 就是运行 storybook 的时候，通过 modernConfig.output.path 来修改

Comment (chenjiahan) [reply packages/cli/plugin-storybook/src/index.ts:30]: 噢噢，如果前置阶段不好判断的话，还是改回上一个写法吧~

Comment (targeral) [reply packages/cli/plugin-storybook/src/index.ts]: changed

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**